test/powershell/Modules/Microsoft.PowerShell.Management/Test-Path.Tests.ps1

```diff
@@ -2,123 +2,136 @@
-	New-Item -Path $testdirectory -Name gatestfile -value 1 -ItemType file | Out-Null
-	New-Item -Path $testdirectory -Name usr -value 1 -ItemType directory | Out-Null
+        # populate with additional files
+        New-Item -Path $testdirectory -Name datestfile -value 1 -ItemType file | Out-Null
+        New-Item -Path $testdirectory -Name gatestfile -value 1 -ItemType file | Out-Null
+        New-Item -Path $testdirectory -Name usr -value 1 -ItemType directory | Out-Null
 
-	$nonExistentDir = Join-Path -Path (Join-Path -Path $testdirectory -ChildPath usr) -ChildPath bin
-	$nonExistentPath = Join-Path -Path (Join-Path -Path (Join-Path -Path $testdirectory -ChildPath usr) -ChildPath bin) -ChildPath error
+        $nonExistentDir = Join-Path -Path (Join-Path -Path $testdirectory -ChildPath usr) -ChildPath bin
+        $nonExistentPath = Join-Path -Path (Join-Path -Path (Join-Path -Path $testdirectory -ChildPath usr) -ChildPath bin) -ChildPath error
     }
 
     It "Should be called on an existing path without error" {
-	{ Test-Path $testdirectory }              | Should -Not -Throw
-	{ Test-Path -Path $testdirectory }        | Should -Not -Throw
-	{ Test-Path -LiteralPath $testdirectory } | Should -Not -Throw
+        { Test-Path $testdirectory }              | Should -Not -Throw
+        { Test-Path -Path $testdirectory }        | Should -Not -Throw
+        { Test-Path -LiteralPath $testdirectory } | Should -Not -Throw
     }
 
     It "Should allow piping objects to it" {
-	{ $testdirectory | Test-Path  } | Should -Not -Throw
+        { $testdirectory | Test-Path  } | Should -Not -Throw
 
-	$testdirectory                  | Test-Path | Should -BeTrue
-	$nonExistentDir                 | Test-Path | Should -BeFalse
+        $testdirectory                  | Test-Path | Should -BeTrue
+        $nonExistentDir                 | Test-Path | Should -BeFalse
     }
 
     It "Should be called on a nonexistent path without error" {
-	{ Test-Path -Path $nonExistentPath } | Should -Not -Throw
+        { Test-Path -Path $nonExistentPath } | Should -Not -Throw
     }
 
     It "Should return false for a nonexistent path" {
-	Test-Path -Path $nonExistentPath | Should -BeFalse
+        Test-Path -Path $nonExistentPath | Should -BeFalse
     }
 
     It "Should return true for an existing path" {
-	{ Test-Path -Path $testdirectory } | Should -BeTrue
+        { Test-Path -Path $testdirectory } | Should -BeTrue
+    }
+
+    It 'Should return false for an empty string' {
+        Test-Path -Path '' | Should -BeFalse
+    }
+
+    It 'Should return false for a whitespace string' {
+        Test-Path -Path '  ' | Should -BeFalse
+    }
+
+    It 'Should write a non-terminating error when given a null path' {
+        { Test-Path -Path $null -ErrorAction Stop }             | Should -Throw -ErrorId 'NullPathNotPermitted'
+        { Test-Path -Path $null -ErrorAction SilentlyContinue } | Should -Not -Throw
```

Comment:
Seems we should remove this. I am not sure that it tests "non-terminating"

Reply:
That's what I thought, too. Surprisingly, though, -ErrorAction SilentlyContinue will still trigger a `| should -throw` if and only if the error is terminating to start with (either `throw` or `ThrowTerminatingError()`)

@indented-automation had to explain that one to me. :)

Reply:
Since this change is explicitly to be a non-terminating error, this pattern seems fine.  But should probably be followed up with `-ErrorAction Stop` to verify the non-terminating error is correct.

Reply:
I'm not sure I follow? I don't see a need to double-check it with -ErrorAction Stop